File: yh_difference/src/yh_difference_client.cpp
#include "ros/ros.h"
#include "yh_difference/YhDifference.h"
#include <cstdlib>

int main(int argc, char** argv)
{
    ros::init(argc, argv, "yh_difference_client");

    if (argc != 3)
    {
        ROS_INFO("rosrun yh_difference yh_difference_client a b");
        ROS_INFO("a, b: int32 number");
        return 1;
    }

    ros::NodeHandle nh;

    ros::ServiceClient client = nh.serviceClient<yh_difference::YhDifference>("yh_difference_service");

    yh_difference::YhDifference srv;
    srv.request.a = atoi(argv[1]);
    srv.request.b = atoi(argv[2]);

    if (client.call(srv))
    {
        ROS_INFO("result: %d", srv.response.result);
    }
    else
    {
        ROS_ERROR("Failed");
        return 1;
    }

    return 0;
}

File: yh_difference/scripts/yh_difference_client.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
from pydoc import cli
import rospy
from yh_difference.srv import YhDifference, YhDifferenceRequest
import sys

def difference_client():
    rospy.init_node("yh_difference_client")
    
    if len(sys.argv) != 3:
        rospy.loginfo("rosrun yh_difference yh_difference_client.py a b")
        rospy.loginfo("a, b: int32 number")
        sys.exit(1)

    client = rospy.ServiceProxy("yh_difference_service", YhDifference)
    req = YhDifferenceRequest()
    req.a = int(sys.argv[1])
    req.b = int(sys.argv[2])

    try:
        res = client(req)
        rospy.loginfo(res.result)
    except rospy.ServiceException:
        rospy.logerr("Failed")
        sys.exit(1)

if __name__ == "__main__":
    difference_client()